docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+    // all data plane types are targeted by the policy.
+    // +kubebuilder:validation:MinItems=1
+    ProxyTypes []TargetRefProxyType `json:"proxyTypes,omitempty"`
+    // Namespace of the referenced resource. If not specified, then equal the value is equal to the resource's namespace
+    // where this TargetRef is used.
+    Namespace string `json:"namespace,omitempty"`
+}
+```
+
+Let's try to review all possible types of references between policies and see if maybe some of them don't make sense.
+We have 3 namespaces: `kuma-system`, `frontend-ns` and `backend-ns`. We have 2 policies: `MeshTimeout`
+and `MeshHTTPRoute`,
+and `MeshTimeout` always references the `MeshHTTPRoute`. Overall, there are 3^2=9 scenarios:
+
+1. MeshTimeout in kuma-system, MeshHTTPRoute in kuma-system
+2. MeshTimeout in kuma-system, MeshHTTPRoute in frontend-ns
+3. MeshTimeout in kuma-system, MeshHTTPRoute in backend-ns
+4. MeshTimeout in frontend-ns, MeshHTTPRoute in kuma-system
+5. MeshTimeout in frontend-ns, MeshHTTPRoute in frontend-ns
+6. MeshTimeout in frontend-ns, MeshHTTPRoute in backend-ns
+7. MeshTimeout in backend-ns, MeshHTTPRoute in kuma-system
+8. MeshTimeout in backend-ns, MeshHTTPRoute in frontend-ns
+9. MeshTimeout in backend-ns, MeshHTTPRoute in backend-ns
+
+Scenarios 2 and 3, 4 and 7, 5 and 9, 6 and 8 are describing the same setup and could be merged.
+We have 5 scenarios left, so let's look closely and try to come up with meaningful use cases for each one.
+
+**1. Both MeshTimeout and MeshHTTPRoute in "kuma-system"**
+
+This scenario is already supported today.
+The use case for this scenario can be the following:
+
+**As** a Backend Service Owner<br>
+**I want** to route all requests with `/v2` prefix to `version: v2` instances<br>
+**so that** requests that need v2 version of the API ended up on correct instances.
+
+**As** a Backend Service Owner<br>
+**I want** all requests routed to `version: v2` instances to have a 5s timeout<br>
+**so that** consumers don't wait longer than max possible processing time.
+
+This can be achieved by applying 2 policies:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: backend-route
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /v2
+          default:
+            backendRefs:
+              - kind: MeshServiceSubset
+                name: backend
+                tags:
+                  version: v2
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: backend-route-timeout
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 5s
+```
+
+**2. MeshTimeout in "kuma-system", MeshHTTPRoute in "frontend-ns"**
```

Comment:
Is there anything in the design that would close up an opportunity to implement this in a future if we were to find use cases for it?

Reply:
Nothing I can think of, we can always add support for this use case